### PR TITLE
JSON layer RFC

### PR DESCRIPTION
**Pull request by ibgreen**
#### Background
- For infovis visualizations from the backend
#### Change List
- New Draft RFC (to be presented and reviewed later)
- RFC directory cleanup

### CODE REVIEW

File: dev-docs/RFCs/v6.1/json-layers-rfc.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+```
+
+
+### Support Declarative Syntax for Accessor-Style functions
+
+In most cases, to meaningfully use deck.gl layers, the user must at least be able to configure accessor functions. Currently deck.gl does not support a fully "declarative" syntax for accessors.
+
+Constant accessors are supported by default, function valued accessors are not. JSON cannot contain functions, however it can contain strings that can be parsed and used to generate functions.
+
+While more advanced formats are possible in the future, the initial idea is to treat strings simplh as object access "paths".
```
Comment (javidhsueh) [inline dev-docs/RFCs/v6.1/json-layers-rfc.md]: typo 'simplh'

File: dev-docs/RFCs/v6.1/json-layers-rfc.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
+# RFC: JSON Layers
+
+* **Authors**: Ib Green
+* **Date**: July 2018
+* **Status**: Draft Proposal
+
+
+## Summary
+
+This RFC proposes adding adaptor classes that accept JSON descriptions of views and layers.
```
Comment (javidhsueh) [inline dev-docs/RFCs/v6.1/json-layers-rfc.md]: What kind of JSON schema would you expect? Something like vega schema?
https://vega.github.io/schema/vega/v2.6.5.json

Comment (ibgreen) [reply dev-docs/RFCs/v6.1/json-layers-rfc.md]: > What kind of JSON schema would you expect? 

Great question! 

In my mind this relates to the "ONE API" principle mentioned at the end of the RFC. Even though deck.gl comes in many incarnations (pure JS, scripting, React, JSON) we want all these APIs to be the "same" or at least similar with very trivial mappings whenever possible. 

That said, we should compare with existing schemas and see how they have solved tricky points, like interactivity etc.

We could of course in addition also support any separate arbitrary schema type(s) we want, but I would really prefer the "core schema" to be very close to the "core API".

File: dev-docs/RFCs/v6.1/json-layers-rfc.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+
+## Background
+
+Especially in the infovis space, there is a growing need to be able to generate powerful visualizations directly from the backend. Being able to describe a visualization in abstract terms and send it to the front-end for display withouth having knowledge about how to code front-end applications can be valuable.
```
Comment (gnavvy) [inline dev-docs/RFCs/v6.1/json-layers-rfc.md]: > Especially in the infovis space, there is a growing need to be able to generate powerful visualizations directly from the backend.

For statistical charts and well-received visualizations such as node-link diagrams, maybe. Basically, we need to know exactly how users want to map their data to the visual channels ahead of time and hardcode the mapping for them in the JSONLayer (unless we can encode the layout algorithms as JSON strings). 

I see the value of prescribed backend-generated visualizations when there's a need for real-time streaming, but I doubt JSON is the right format for that.

Regarding declarative APIs for InfoVis, there's already good [studies](https://vega.github.io/vega-lite/examples/) on which types of visualizations can be defined declaratively. I agree we should keep our declarative interface as close as possible to other incarnations of ours, but we need to design carefully not to reject users switching from the vega community.

I guess we deprioritized the declarative API last time due to we could not find a good user cohort for it. Or, who will use this JSONLayer? For those who want to get their hand dirty, they can just code in JS and use our layers directly. For users who just want the capability, we build workflow tools for them. If we can identify good use cases for the backend generated InfoVis (like streaming pre-layout node-link graph from backend via Apache Arrow), I'm totally for it.

**Pull request merged**